### ISSUE TITLE
`pulumi login` returns 403 for `azblob`

### ISSUE THREAD

**Issue by jjcz1** [pulumi/pulumi#21878]
### What happened?

- `pulumi login azblob://pulumi?storage_account=mystorage` always fails with
```
error: problem logging in: read ".pulumi\\meta.yaml": blob (key ".pulumi/meta.yaml") (code=Unknown): GET https://mystorage.blob.core.windows.net/pulumi/.pulumi/meta.yaml
--------------------------------------------------------------------------------
RESPONSE 403: 403 This request is not authorized to perform this operation using this permission.
ERROR CODE: AuthorizationPermissionMismatch
```

- I have `Contributor` and `Storage Blob Data Contributor` roles, `az` commands reading / writing data in this container work fine.
- This error persist even after `az logout`.
- I don't have any other env variables set, or other cached credentials AFAIK.

From what I understood from the docs, I should be able authenticate locally w/o setting any env vars, or am I mistaken here?

### Example

```pwsh
az login
pulumi login azblob://pulumi?storage_account=mystorage
```

or

```pwsh
az login
$env:AZURE_STORAGE_ACCOUNT=mystorage
pulumi login azblob://pulumi
```

### Output of `pulumi about`

```
CLI          
Version      3.221.0
Go Version   go1.26.0
Go Compiler  gc

Host     
OS       Microsoft Windows 10 Enterprise
Version  10.0.19045 Build 19045
Arch     x86_64
```
